### PR TITLE
fix(css_semantic): semantic event for css supports at-rule

### PR DESCRIPTION
**Pull request by mehm8128**
## Summary

fixed css semantic event so that `noDuplicateProperties` properly throws lint error when we use `@supports`.

close https://github.com/biomejs/biome/issues/4756

## Test Plan

Please check snap shot test

### CODE REVIEW

File: crates/biome_css_semantic/src/events.rs [hunk 1]
```diff
@@ -65,7 +65,8 @@ impl SemanticEventExtractor {
             // allowing for proper scoping and inheritance of styles.
             kind if kind == CSS_QUALIFIED_RULE
                 || kind == CSS_NESTED_QUALIFIED_RULE
-                || kind == CSS_MEDIA_AT_RULE =>
+                || kind == CSS_MEDIA_AT_RULE
+                || kind == CSS_SUPPORTS_AT_RULE =>
```
Comment (mehm8128) [inline crates/biome_css_semantic/src/events.rs:69]: fixed double thrown diagnostics adding this.

ref: https://github.com/biomejs/biome/issues/4756#issuecomment-2551518226
>Plus, the rule signals both bindings, while it should signal only one. (There are two diagnostics that basically say the same thing)

File: crates/biome_css_semantic/src/events.rs [hunk 2]
```diff
@@ -95,6 +96,10 @@ impl SemanticEventExtractor {
                     .for_each(|s| self.process_selector(s));
             }
             CSS_DECLARATION => {
+                if matches!(node.parent().kind(), Some(CSS_SUPPORTS_FEATURE_DECLARATION)) {
+                    return;
+                }
```
Comment (mehm8128) [inline crates/biome_css_semantic/src/events.rs:101]: `@supports` uses `CSS_SUPPORTS_FEATURE_DECLARATION` for writing the condition but this is wrapping the `CSS_DECLARATION` here (`parse_declaration`): https://github.com/biomejs/biome/blob/f9ff72d3f58c24db73fa84967eea489ce8492839/crates/biome_css_parser/src/syntax/at_rule/supports.rs#L164-L177

so if its parent is `CSS_SUPPORTS_FEATURE_DECLARATION`, I returned without following processes.
But I don't completely understand these processes, so if there is other better approaches let me know.

File: crates/biome_css_analyze/tests/specs/nursery/noDuplicateProperties/valid.css [hunk 3]
```diff
@@ -15,6 +15,12 @@ a { color: pink; @media { color: orange; &::before { color: black; } } }
 
 a { --custom-property: 0; --custom-property: 1; }
 
+a {
+    @supports (color: pink) {
+        color: pink;
+    }
+}
+
```
Comment (dyc3) [inline crates/biome_css_analyze/tests/specs/nursery/noDuplicateProperties/valid.css:23]: This looks good to me as a positive test, but I'd like to see a negative test case as well. FYI, you can add other files in this folder, you aren't necessarily limited to `valid.css` and `invalid.css`

Comment (mehm8128) [reply crates/biome_css_analyze/tests/specs/nursery/noDuplicateProperties/valid.css:23]: Thank you! I added more tests for both valid and invalid cases, and I confirmed that the `noDuplicateProperties` lint rule properly throws errors for invalid test cases. Please check it again!

File: crates/biome_css_analyze/tests/specs/nursery/noDuplicateProperties/invalid.css.snap [hunk 4]
```diff
@@ -49,337 +49,408 @@ a { color: red !important; color: blue; }
...
+    60 │     }
+
+  i color is already defined here.
+
+    55 │         color: pink;
+    56 │         &:hover {
+  > 57 │             color: orange;
+       │             ^^^^^
+    58 │             color: black;
+    59 │         }
```
Comment (mehm8128) [inline crates/biome_css_analyze/tests/specs/nursery/noDuplicateProperties/invalid.css.snap:451]: it seems that the lint rule properly throws errors for duplicated CSS properties

File: crates/biome_css_analyze/tests/specs/nursery/noDuplicateProperties/invalid.css.snap [hunk 5]
```diff
@@ -378,6 +396,59 @@ invalid.css:44:28 lint/nursery/noDuplicateProperties ━━━━━━━━━
...
+    60 │     }
+  
+  i color is already defined here.
+  
+    55 │         color: pink;
+    56 │         &:hover {
+  > 57 │             color: orange;
+       │             ^^^^^
+    58 │             color: black;
+    59 │         }
```
Comment (mehm8128) [inline crates/biome_css_analyze/tests/specs/nursery/noDuplicateProperties/invalid.css.snap:451]: it seems that the lint rule properly throws diagnostics for duplicated CSS properties😀

**Review by dyc3 — APPROVED** [review]

**Pull request merged**